RecognizeDtmf/RecognizeDtmf.cs

```diff
@@ -0,0 +1,327 @@
+            reportCancellationToken.Register(() => playAudioCompletedTask.TrySetCanceled());
+
+            var playCompletedNotification = new NotificationCallback((callEvent) =>
+            {
+                Task.Run(() =>
+                {
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, $"Play audio status: Completed");
+                    playAudioCompletedTask.TrySetResult(true);
+                    EventDispatcher.Instance.Unsubscribe("PlayCompleted", callConnectionId);
+                });
+            });
+
+            var playFailedNotification = new NotificationCallback((callEvent) =>
+            {
+                Task.Run(() =>
+                {
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, $"Play audio status: Failed");
+                    playAudioCompletedTask.TrySetResult(false);
+                    EventDispatcher.Instance.Unsubscribe("PlayFailed", callConnectionId);
+                });
+            });
+
+            var playCancelledNotification = new NotificationCallback((callEvent) =>
+            {
+                Task.Run(() =>
+                {
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, $"Play audio status: Cancelled");
+                    playAudioCompletedTask.TrySetResult(false);
+                    EventDispatcher.Instance.Unsubscribe("PlayCancelled", callConnectionId);
+                });
+            });
+
+            //Subscribe to event
+            EventDispatcher.Instance.Subscribe("PlayCompleted", callConnectionId, playCompletedNotification);
+            EventDispatcher.Instance.Subscribe("PlayFailed", callConnectionId, playFailedNotification);
+            EventDispatcher.Instance.Subscribe("PlayCancelled", callConnectionId, playCancelledNotification);
+        }
+
+        private void RegisterToDtmfResultEvent(string callConnectionId)
+        {
+            recognizeDtmfCompletedTask = new TaskCompletionSource<bool>(TaskContinuationOptions.RunContinuationsAsynchronously);
+
+            var dtmfReceivedEvent = new NotificationCallback((callEvent) =>
+            {
+                Task.Run(() =>
+                {
+                    var toneReceivedEvent = (RecognizeCompleted)callEvent;
+
+                    if (toneReceivedEvent.CollectTonesResult.Tones.Count != 0)
+                    {
+                        Logger.LogMessage(Logger.MessageType.INFORMATION, $"Tone received --------- : {toneReceivedEvent.CollectTonesResult.Tones[0]}");
+                        toneCount = toneReceivedEvent.CollectTonesResult.Tones.Count;
+                        toneInputValue = toneReceivedEvent.CollectTonesResult.Tones[0];
+                    }
+                    EventDispatcher.Instance.Unsubscribe("RecognizeCompleted", callConnectionId);
+                    recognizeDtmfCompletedTask.TrySetResult(true);
+                });
+            });
+
+            var dtmfFailedEvent = new NotificationCallback((callEvent) =>
```

Comment:
TypeCast this to RecognizeFailed event as above you are doing for RecognizeCompleted event

Reply:
Please do the same for PlayEvents as well